lib/sinon/test.js

```diff
@@ -55,6 +55,15 @@
                     sandbox.verifyAndRestore();
                 }
 
+                if (result && typeof result.then === "function" && typeof result.catch === "function") {
```

Comment:
This change will only make our Promise duck-typing more permissive. It also conforms to the Promise/A+ checking for promises. 

https://promisesaplus.com/